lib/matplotlib/table.py

```diff
@@ -743,6 +744,11 @@ def table(ax,
         rows = len(cellColours)
         cols = len(cellColours[0])
         cellText = [[''] * cols] * rows
+    # Convert the Pandas dataframe to a 2d string array
+    elif isinstance(cellText, pd.DataFrame):
```

Comment:
Because pandas is an optional dependency, we cannot `isinstance` it here. The typical approach would be duck-typing, see 

https://github.com/matplotlib/matplotlib/blob/64d45cb8f45a38109b729cd3e1835e2dd36e65fc/lib/matplotlib/cbook.py#L2395

but since you need the column headers, that may not be sufficient here. It think we have to go though someting analogous to 

https://github.com/matplotlib/matplotlib/blob/64d45cb8f45a38109b729cd3e1835e2dd36e65fc/lib/matplotlib/cbook.py#L2356

Reply:
Following your idea, I implemented a _is_pandas_dataframe(x) method and modified unpack_to_numpy to include it. I used these two methods inside the table.py instead and validated the change with a unit test.